yt_dlp/extractor/crunchyroll.py

```diff
@@ -733,13 +736,111 @@ class CrunchyrollBetaIE(CrunchyrollBaseIE):
+            display_id)
+        client_id = traverse_obj(app_config, ('cxApiParams', 'accountAuthClientId'))
+        api_domain = traverse_obj(app_config, ('cxApiParams', 'apiDomain'))
+        basic_token = str(base64.b64encode(('%s:' % client_id).encode('ascii')), 'ascii')
+        auth_response = self._download_json(
+            f'{api_domain}/auth/v1/token', display_id,
+            note='Authenticating with cookie',
+            headers={
+                'Authorization': 'Basic ' + basic_token
+            }, data='grant_type=etp_rt_cookie'.encode('ascii'))
+        policy_response = self._download_json(
+            api_domain + '/index/v2', display_id,
+            note='Retrieving signed policy',
+            headers={
+                'Authorization': auth_response.get('token_type') + ' ' + auth_response.get('access_token')
+            })
+        bucket = traverse_obj(policy_response, ('cms', 'bucket'))
+        params = {
+            'Policy': traverse_obj(policy_response, ('cms', 'policy')),
+            'Signature': traverse_obj(policy_response, ('cms', 'signature')),
+            'Key-Pair-Id': traverse_obj(policy_response, ('cms', 'key_pair_id')),
+            'locale': traverse_obj(initial_state, ('localization', 'locale'))
+        }
+        episode_response = self._download_json(
+            api_domain + '/cms/v2' + bucket + '/episodes/' + internal_id, display_id,
+            note='Retrieving episode metadata',
+            query=params)
+        stream_response = self._download_json(
+            episode_response.get('playback'), display_id,
+            note='Retrieving stream info')
+
+        thumbnails = []
+        for thumbnails_data in traverse_obj(episode_response, ('images', 'thumbnail')):
+            for thumbnail_data in thumbnails_data:
+                thumbnails.append({
+                    'url': thumbnail_data.get('source'),
+                    'width': thumbnail_data.get('width'),
+                    'height': thumbnail_data.get('height'),
+                })
+        subtitles = {}
+        for lang, subtitle_data in stream_response.get('subtitles').items():
+            subtitles[lang] = [{
+                'url': subtitle_data.get('url'),
+                'ext': subtitle_data.get('format')
+            }]
+
+        requested_languages = self._configuration_arg('language')
+        requested_hardsubs = [('' if val == 'none' else val) for val in self._configuration_arg('hardsub')]
+        language_preference = qualities((requested_languages or [traverse_obj(initial_state, ('localization', 'locale')) or ''])[::-1])
+        hardsub_preference = qualities((requested_hardsubs or ['', traverse_obj(initial_state, ('localization', 'locale')) or ''])[::-1])
+
+        formats = []
+        for stream_type, streams in stream_response.get('streams', {}).items():
+            if stream_type in ('adaptive_hls'):
+                for stream in streams.values():
```

Comment:
```suggestion
                for stream in streams.values():
                    audio_lang = stream_response.get('audio_locale') or ''
                    hardsub_lang = stream_response.get('hardsub_locale') or ''
                    if (requested_languages and audio_lang.lower() not in requested_languages
                            or requested_hardsubs and hardsub_lang.lower() not in requested_hardsubs):
                        continue
```
I believe this is what you want to reduce requests

Reply:
One potential wrinkle here. The beta api uses language tags with dashes (`en-US`), rather than without (`enUS`) like the old system. Should we provide compatibility with the extractor args for crunchyroll here or match the api?

For that matter, should we be using the same extractor args as `CrunchyrollIE` or use a separate key for beta?

Reply:
Imo, it is best to use the same key and maintain compat since it's just a matter of doing `.replace('-', '')`.

But I'll leave the decision to your discretion. I dont use the extractor, and you are in a better position to judge whether users would want to apply the same key to it or want the flexibility of separate keys

Reply:
In that case, the language codes used for the extractor args wouldn't match those used for subtitles, unless we converted those too.

Which I'm against, since the new system follows standards (RFC 5646), so we probably shouldn't be porting back to non-standardized tags just for backward compatibility.

Using yt-dlp for anime downloading across multiple sites already involves handling the language tags on a per-site basis, so I don't think changing them as part of the beta site handling is a big hassle to users.

Also, the `language` extractor arg from `CrunchyrollIE` doesn't actually apply here, as audio language is fixed (at an api level) for a given video id (dubbed versions of content fall under their own season ids and episode ids), and it makes sense to implement an extractor arg to choose which stream types to retrieve, so I think I'm going to use a separate key string for beta.